pkg/cloudprovider/vsphere/nodemanager.go

```diff
@@ -330,7 +332,13 @@ func (nm *NodeManager) DiscoverNode(nodeID string, searchBy cm.FindVM) error {
 				} else if !foundInternal && foundExternal {
 					klog.Warning("External address found, but internal address not found. Returning what addresses were discovered.")
 				}
-				break
+				if _, ok := os.LookupEnv("ENABLE_ALPHA_DUAL_STACK"); ok {
```

Comment:
Would prefer the check for `ENABLE_ALPHA_DUAL_STACK` to happen in the same place we check the NSX-T gate in `buildVSphereFromConfig`:

https://github.com/kubernetes/cloud-provider-vsphere/blob/22e85632bfe62f9a6afff8d2fc269f80eb9e7c78/pkg/cloudprovider/vsphere/cloud.go#L179-L211

Reply:
`DiscoverNode` in `nodemanager.go` should assume dualstack if  `ipFamily` has multiple entries. But configuration of `ipFamily` should be gated by `ENABLE_ALPHA_DUAL_STACK`.

Reply:
yeah, that makes sense. So, how does the ipFamily look like when `ENABLE_ALPHA_DUAL_STACK` is disabled. Should ipfamily just become a one-entry array(i.e. leaving only the first entry in the array)?

Reply:
I think if `ENABLE_ALPHA_DUAL_STACK` is disabled but the config has multiple IP families, we should treat that as an error and exit. What do you think?

Reply:
I think what you proposed has a better behavior and informs the user to provide the correct ipfamily. Going to get started on that direction.